src/Application.cpp

```diff
@@ -392,6 +392,70 @@ namespace weatherserver {
     location.setIsAddingWeatherToAddressSpace(false);
   }
 
+  /*
+  Checking if locations number in the server model matches with passed value. Trying to adjust if its different.
+  Returns true if:
+      - model locations number was successfuly read and already matches with passed value;
+      - value is different, but adjustment (write) was successful.
+  Returns false if:
+      - couldn't read existing value from the model;
+      - existing value was different and adjustment (write) failed.
+  */
+  static bool validateLocationsNumberInTheModel(UA_Server& server, const CountryData& country, uint32_t valueToValidate) {
+
+    bool validationFlag = false;
```

Comment:
Using flag like this one is very in C-style (contrary to C++ style where we just return from the function when needed). But here we are using C library and its probably ok.

For C++ there is for example [Boost.ScopeExit](https://www.boost.org/doc/libs/1_71_0/libs/scope_exit/doc/html/index.html#scope_exit.introduction) which is possible to use for freeing memory allocated inside `UA_Variant valueInModel`.

Reply:
Ok. I'll make a note of that.